test/plugins/test_convert.py

```diff
@@ -147,9 +147,9 @@ def run_convert_path(self, path, *args):
         """Run the `convert` command on a given path."""
         # The path is currently a filesystem bytestring. Convert it to
         # an argument bytestring.
-        path = os.fsencode(os.fsdecode(path))
+        path = os.fsdecode(path)
 
-        args = args + (b"path:" + path,)
+        args = args + ("path:" + path,)
```

Comment:
This should simplify to

```py
def run_convert_path(self, path, *args):
    """Run the `convert` command on a given path."""
    return self.run_command("convert", *args, b"path:" + path)
```

unless of course, we don't actually call it with `path: bytes` consistently. In that case, the `fsdecode` should stay.

Reply:
Nice cleanup! Unfortunately `run_command` does not support `bytes` arguments, so this needs to be a string:

```python
    def run_convert_path(self, path, *args):
        """Run the `convert` command on a given path."""
        return self.run_command("convert", *args, f"path:{os.fsdecode(path)}")
```

Reply:
Ah, sorry, must have mixed it up. Thanks for fixing up my suggestion :)